.spin/cmds.py

```diff
@@ -0,0 +1,1065 @@
+        elif release:
+            buildtype = 'release'
+            cflags_unwanted = ('-O0', '-O1', '-O2')
+        meson_args = meson_args + (f"-Dbuildtype={buildtype}", )
+        if 'CFLAGS' in os.environ.keys():
+            # Check that CFLAGS doesn't contain something that supercedes -O0
+            # for a plain debug build (conda envs tend to set -O2)
+            cflags = os.environ['CFLAGS'].split()
+            for flag in cflags_unwanted:
+                if flag in cflags:
+                    raise ValueError(f"A {buildtype} build isn't possible, "
+                                        f"because CFLAGS contains `{flag}`."
+                                        "Please also check CXXFLAGS and FFLAGS.")
+
+    if asan:
+        meson_args = meson_args + ('-Db_sanitize=address,undefined', )
+
+    if setup_args:
+        meson_args = meson_args + tuple([str(arg) for arg in setup_args])
+
+    if with_accelerate:
+        # on a mac you probably want to use accelerate over scipy_openblas
+        meson_args = meson_args + ("-Dblas=accelerate", )
+    elif with_scipy_openblas:
+        configure_scipy_openblas()
+        os.environ['PKG_CONFIG_PATH'] = os.pathsep.join([
+                os.getcwd(),
+                os.environ.get('PKG_CONFIG_PATH', '')
+                ])
+
+    if parallel is None:
+        # Use number of physical cores rather than ninja's default of 2N+2,
+        # to avoid out of memory issues (see gh-17941 and gh-18443)
+        n_cores = cpu_count(only_physical_cores=True)
+        jobs = n_cores
+    else:
+        jobs = parallel
+
+    meson_install_args = meson_install_args + ("--tags=" + tags, )
+
+    if show_build_log:
+        verbose = show_build_log
+
+    parent_callback(**{MESON_ARGS: meson_args,
+                       MESON_COMPILE_ARGS: meson_compile_args,
+                       MESON_INSTALL_ARGS: meson_install_args,
+                       "jobs": jobs,
+                       "verbose": verbose,
+                       **kwargs})
+
+@click.option(
+    '--durations', '-d', default=None, metavar="NUM_TESTS",
+    help="Show timing for the given number of slowest tests"
+)
+@click.option(
+    '--submodule', '-s', default=None, metavar='MODULE_NAME',
+    help="Submodule whose tests to run (cluster, constants, ...)")
+@click.option(
+    '--mode', '-m', default='not slow', metavar='MODE', show_default=True,
+    help=("'fast', 'full', or something that could be passed to "
+            "`pytest -m` as a marker expression"))
+@click.option(
+    '--parallel', '-j', default=1, metavar='N_JOBS',
+    help="Number of parallel jobs for testing"
+)
+@click.option(
+    '--array-api-backend', '-b', default=None, metavar='ARRAY_BACKEND',
+    multiple=True,
+    help=(
+        "Array API backend "
+        "('all', 'numpy', 'torch', 'cupy', 'array_api_strict', 'jax.numpy')."
+    )
+)
+@spin.util.extend_command(spin.cmds.meson.test)
+def test(*, parent_callback, pytest_args, tests, coverage,
```

Comment:
While we're at it - https://github.com/scientific-python/specs/pull/324?
```suggestion
def tests(*, parent_callback, pytest_args, tests, coverage,
```

Reply:
Please don't do that, it will mess with muscle memory too much. It won't make the cut in the SPEC anyway I think, Henry's comment that `test` is way more common is correct.

The `spin` defaults seem fine here.

Reply:
Okay, that's fair.

> Henry's comment that `test` is way more common is correct.

I believe he was referring to extras rather than tasks there, but maybe we have different comments in mind.